sdk/identity/azure-identity/azure/identity/_credentials/workload_identity.py

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import Any
+import logging
+from azure.core.credentials import AccessToken
+from .token_exchange import TokenExchangeCredential
+from .._internal.decorators import log_get_token
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WorkloadIdentityCredential:
+    """WorkloadIdentityCredential supports Azure workload identity on Kubernetes.
+    See https://learn.microsoft.com/azure/aks/workload-identity-overview for more information
+
+    :param str tenant_id: ID of the application's Azure Active Directory tenant. Also called its "directory" ID.
+    :param str client_id: the client ID of an Azure AD app registration.
+    :param str file: The path to a file containing a Kubernetes service account token that authenticates the identity.
+    """
+
+    def __init__(
+            self,
+            tenant_id: str,
+            client_id: str,
+            file: str,
+            **kwargs: Any
+    ) -> None:
+        kwargs.pop("token_file_path", None)
+        self._credential = TokenExchangeCredential(
```

Comment:
Would be simpler to replace `TokenExchangeCredential` instead? Keeping it seems to add an unnecessary layer to this credential.